sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -155,6 +154,8 @@ case class CreateDataSourceTableAsSelectCommand(
       } else {
         table.storage.locationUri
       }
+
+      sparkSession.sessionState.catalog.checkTableOrPathExists(table, ignoreIfExists = false)
```

Comment:
because in following `saveDataIntoTable` will create the table path and store data in it, we should not check the path in following `createTable`, we check it here before `saveDataIntoTable`